doc/developer-guides/coding_guidelines.rst

```diff
@@ -1121,6 +1428,312 @@ Compliant example::
        struct struct_showcase showcase = {32U, -1};
 
 
+TY-09: The type used in switch statement shall be consistent
+============================================================
+
+The type shall be consistent between the case expression and the controlling
+expression of switch statement.
+
+Compliant example::
+
+    enum enum_showcase {
+        ENUM_SHOWCASE_0,
+        ENUM_SHOWCASE_1,
+        ENUM_SHOWCASE_2
+    };
+    
+    enum enum_showcase showcase;
+    
+    switch (showcase) {
+    case ENUM_SHOWCASE_0:
+        /* showcase */
+        break;
+    case ENUM_SHOWCASE_1:
+        /* showcase */
+        break;
+    default:
+        /* showcase */
+        break;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       enum enum_showcase {
+           ENUM_SHOWCASE_0,
+           ENUM_SHOWCASE_1,
+           ENUM_SHOWCASE_2
+       };
+       
+       enum enum_showcase showcase;
+       
+       switch (showcase) {
+       case ENUM_SHOWCASE_0:
+           /* showcase */
+           break;
+       case 1U:
+           /* showcase */
+           break;
+       default:
+           /* showcase */
+           break;
+
+
+TY-10: const qualifier shall not be discarded in cast operation
+===============================================================
+
+Compliant example::
+
+    const uint32_t *showcase_const;
+    const uint32_t *showcase = showcase_const;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       const uint32_t *showcase_const;
+       uint32_t *showcase = (uint32_t *)showcase_const;
+
+
+TY-11: A variable shall be declared as static if it is only used in one file
+============================================================================
+
+Compliant example::
+
+    /* In `showcase.c`: */
+    /* `showcase` is only in `showcase.c` */
+    static uint32_t showcase;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* In `showcase.c`: */
+       /* `showcase` is only in `showcase.c` */
+       uint32_t showcase;
+
+
+TY-12: The type conversion shall be explicit
+============================================
+
+Compliant example::
+
+    uint32_t showcase_u32;
+    uint64_t showcase_u64 = 64UL;
+    
+    showcase_u32 = (uint32_t)showcase_u64;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_u32;
+       uint64_t showcase_u64 = 64UL;
+       
+       showcase_u32 = showcase_u64;
+
+
+TY-13: A complex integer expression shall not be cast to a wider type
+=====================================================================
+
```

Comment:
Maybe: ``Cast operands, not arithmetic expressions`` and it would replace TY-13, 14, and 15.
You could also merge the compliant and non-compiant examples.